packages/clay-pagination-bar/src/PaginationBarWithBasicItems.tsx

```diff
@@ -152,6 +152,14 @@ export const ClayPaginationBarWithBasicItems: React.FunctionComponent<IProps> =
 		return item;
 	});
 
+	const totalPages = Math.ceil(totalItems / activeDelta);
+
+	React.useEffect(() => {
+		if (onPageChange && activePage > totalPages) {
+			onPageChange(1);
+		}
+	}, [totalPages]);
```

Comment:
I don't know this component well enough to know if this is The Best Fix™, but based on my superficial understanding it looks like it will fix it.